File: game_client/Assets/Scripts/ClientProtocol/GS2GC/p036_TreasureHuntOp/GS2GC_036_010_RetTreasureHuntTransOreRankInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p036_TreasureHuntOp
{

public class GS2GC_036_010_RetTreasureHuntTransOreRankInfo : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 矿石排行榜前三
/// </summary>
private List<Common.TreasureHuntObj.TreasureHunt_OreRankItem> topThreeList;
/// <summary>
/// 当前排名
/// </summary>
private int rank;


public GS2GC_036_010_RetTreasureHuntTransOreRankInfo() {
	topThreeList = new List<Common.TreasureHuntObj.TreasureHunt_OreRankItem>();
	rank = 0;
}

public GS2GC_036_010_RetTreasureHuntTransOreRankInfo(
	List<Common.TreasureHuntObj.TreasureHunt_OreRankItem> _topThreeList
	, int _rank
) {	topThreeList = _topThreeList;
	rank = _rank;
}

public byte getMainOrder() { return (byte)36; }

public byte getSubOrder() { return (byte)10; }

/// <summary>
/// 矿石排行榜前三
/// </summary>
public List<Common.TreasureHuntObj.TreasureHunt_OreRankItem> getTopThreeList() { return topThreeList; }
/// <summary>
/// 矿石排行榜前三
/// </summary>
public void addTopThreeList(Common.TreasureHuntObj.TreasureHunt_OreRankItem _topThreeList) { topThreeList.Add(_topThreeList); }
/// <summary>
/// 当前排名
/// </summary>
public int getRank() { return rank; }
/// <summary>
/// 当前排名
/// </summary>
public void setRank(int _rank) { rank = _rank; }


public int GetBufSize() {
	int _size = 4;
	_size += 2 + (topThreeList.Count * 20);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 6;
	_size += 2 + (topThreeList.Count * 20);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _topThreeListCount = _buf.getShort();
	for(int _i = 0; _i < _topThreeListCount; _i++) { 
		Common.TreasureHuntObj.TreasureHunt_OreRankItem _topThreeList = new Common.TreasureHuntObj.TreasureHunt_OreRankItem();
		int __topThreeListCustLen = _buf.getInt();
	int __topThreeListCurPos = _buf.getCurPos();
	_topThreeList.ReadUnzipBuf(_buf, __topThreeListCurPos + __topThreeListCustLen);
	_buf.setPosition(__topThreeListCurPos + __topThreeListCustLen);

		topThreeList.Add(_topThreeList);
	}
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	rank = _buf.getInt();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putShort((short)topThreeList.Count);
	for(int _i = 0; _i < topThreeList.Count; _i++) { 
		_buf.putInt(topThreeList[_i].GetBufSize());
	topThreeList[_i].PutUnzipBuf(_buf);
	}
	_buf.putInt(rank);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)36);
	_buf.put((byte)10);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)36);
	_recBuf.put((byte)10);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("topThreeList").Append(":").Append(topThreeList.ToString()).Append(", ");
	builder.Append("rank").Append(":").Append(rank.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ServerProtocol/src/GS2GC/p036_TreasureHuntOp/GS2GC_036_010_RetTreasureHuntTransOreRankInfo.java
package GS2GC.p036_TreasureHuntOp;

import java.nio.ByteBuffer;
public class GS2GC_036_010_RetTreasureHuntTransOreRankInfo implements ALBasicProtocolPack._IALProtocolStructure {
/** 矿石排行榜前三 */
private java.util.ArrayList<Common.TreasureHuntObj.TreasureHunt_OreRankItem> topThreeList;
/** 当前排名 */
private int rank;


public GS2GC_036_010_RetTreasureHuntTransOreRankInfo() {
	topThreeList = new java.util.ArrayList<Common.TreasureHuntObj.TreasureHunt_OreRankItem>();
	rank = 0;
}

public GS2GC_036_010_RetTreasureHuntTransOreRankInfo(
	 java.util.ArrayList<Common.TreasureHuntObj.TreasureHunt_OreRankItem> _topThreeList
	, int _rank
) {	topThreeList = _topThreeList;
	rank = _rank;
}

public final byte getMainOrder() { return (byte)36; }

public final byte getSubOrder() { return (byte)10; }

/** 矿石排行榜前三 */
public java.util.ArrayList<Common.TreasureHuntObj.TreasureHunt_OreRankItem> getTopThreeList() { return topThreeList; }
/** 矿石排行榜前三 */
public void addTopThreeList(Common.TreasureHuntObj.TreasureHunt_OreRankItem _topThreeList) { topThreeList.add(_topThreeList); }
/** 当前排名 */
public int getRank() { return rank; }
/** 当前排名 */
public void setRank(int _rank) { rank = _rank; }


public final int GetBufSize() {
	int _size = 4;
	_size += 2 + (topThreeList.size() * 20);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 6;
	_size += 2 + (topThreeList.size() * 20);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _topThreeListCount = _buf.getShort();
	for(int _i = 0; _i < _topThreeListCount; _i++) { 
		Common.TreasureHuntObj.TreasureHunt_OreRankItem _topThreeList = new Common.TreasureHuntObj.TreasureHunt_OreRankItem();
		if(_buf.remaining() <= 0) return;
	int __topThreeListCustLen = _buf.getInt();
	int __topThreeListCurPos = _buf.position();
	_topThreeList.ReadUnzipBuf(_buf, __topThreeListCurPos + __topThreeListCustLen);
	_buf.position(__topThreeListCurPos + __topThreeListCustLen);

		topThreeList.add(_topThreeList);
	}
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) rank = _buf.getInt();
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putShort((short)topThreeList.size());
	for(int _i = 0; _i < topThreeList.size(); _i++) { 
		_buf.putInt(topThreeList.get(_i).GetBufSize());
	topThreeList.get(_i).PutUnzipBuf(_buf);
	}
	_buf.putInt(rank);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)36);
	_buf.put((byte)10);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)36);
	_recBuf.put((byte)10);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

